src/main/java/com/mineboundteam/minebound/inventory/AlloyFurnaceMenu.java

```diff
@@ -0,0 +1,225 @@
+        super(pMenuType, pContainerId);
+        this.recipeType = pRecipeType;
+        this.recipeBookType = pRecipeBookType;
+        checkContainerSize(pContainer, 3);
+        checkContainerDataCount(pData, 4);
+        this.container = pContainer;
+        this.data = pData;
+        this.level = pPlayerInventory.player.level;
+
+        addSlots(pPlayerInventory);
+
+        addDataSlots(pData);
+    }
+
+    public AlloyFurnaceMenu(int containerId, Inventory playerInventory, FriendlyByteBuf buf) {
+        this(containerId, playerInventory, new SimpleContainer(5), new SimpleContainerData(4));
+    }
+
+    protected void addSlots(Inventory pPlayerInventory){
+        this.addSlot(new Slot(container, 0, 22, 34));
+        this.addSlot(new Slot(container, 3, 45, 21));
+        this.addSlot(new Slot(container, 4, 68, 34));
+        this.addSlot(new FuckingFuelSlot(this, container, 1, 45, 62));
+        this.addSlot(new FurnaceResultSlot(pPlayerInventory.player, container, 2, 139, 34));
+
+        addInventory(pPlayerInventory);
+        addHotBar(pPlayerInventory);
+    }
+
+    protected void addHotBar(Inventory playerInventory) {
+        for (int i = 0; i < 9; ++i) {
+            this.addSlot(new Slot(playerInventory, i, 8 + i * 18, 144));
+        }
+    }
+
+    protected void addInventory(Inventory playerInventory) {
+        for (int i = 0; i < 3; ++i) {
+            for (int j = 0; j < 9; ++j) {
+                this.addSlot(new Slot(playerInventory, j + i * 9 + 9, 8 + j * 18, 86 + i * 18));
+            }
+        }
+    }
+
+    @Override
+    public ItemStack quickMoveStack(Player pPlayer, int pIndex) {
+        ItemStack itemstack = ItemStack.EMPTY;
+        Slot slot = this.slots.get(pIndex);
+        if (slot != null && slot.hasItem()) {
+            ItemStack itemstack1 = slot.getItem();
+            itemstack = itemstack1.copy();
+            if (pIndex == 4) {
+                if (!this.moveItemStackTo(itemstack1, 5, 41, true)) {
+                    return ItemStack.EMPTY;
+                }
+                slot.onQuickCraft(itemstack1, itemstack);
+            } else if (pIndex == 3 && this.canSmelt(itemstack1)){
+                if (!this.moveItemStackTo(itemstack1, 0, 3, false)) {
+                    return ItemStack.EMPTY;
+                }
+            } else if (pIndex > 4) {
+                if (this.isFuel(itemstack1)) {
+                    if (!this.moveItemStackTo(itemstack1, 3, 4, false)) {
+                        return ItemStack.EMPTY;
+                    }
+                } else if (this.canSmelt(itemstack1)) {
+                    if (!this.moveItemStackTo(itemstack1, 0, 3, false)) {
+                        return ItemStack.EMPTY;
+                    }
+                } else if (pIndex >= 5 && pIndex < 32) {
```

Comment:
`pIndex >= 5` is always true since it is nested inside an `if (pIndex > 4)`, which are the same thing, can be removed.